src/coreComponents/physicsSolvers/fluidFlow/ThermalSinglePhaseFVMKernels.hpp

```diff
@@ -476,6 +476,353 @@ class FaceBasedAssemblyKernelFactory
+ */
+template< integer NUM_DOF, typename FLUIDWRAPPER >
+class DirichletFaceBasedAssemblyKernel : public singlePhaseFVMKernels::DirichletFaceBasedAssemblyKernel< NUM_DOF, FLUIDWRAPPER >
+{
+public:
+
+/**
+ * @brief The type for element-based data. Consists entirely of ArrayView's.
+ *
+ * Can be converted from ElementRegionManager::ElementViewConstAccessor
+ * by calling .toView() or .toViewConst() on an accessor instance
+ */
+  template< typename VIEWTYPE >
+  using ElementViewConst = ElementRegionManager::ElementViewConst< VIEWTYPE >;
+
+  using AbstractBase = singlePhaseFVMKernels::FaceBasedAssemblyKernelBase;
+  using DofNumberAccessor = AbstractBase::DofNumberAccessor;
+  using PermeabilityAccessors = AbstractBase::PermeabilityAccessors;
+  using SinglePhaseFlowAccessors = AbstractBase::SinglePhaseFlowAccessors;
+  using SinglePhaseFluidAccessors = AbstractBase::SinglePhaseFluidAccessors;
+
+  using AbstractBase::m_dt;
+  using AbstractBase::m_rankOffset;
+  using AbstractBase::m_dofNumber;
+  using AbstractBase::m_ghostRank;
+  using AbstractBase::m_gravCoef;
+  using AbstractBase::m_mob;
+  using AbstractBase::m_pres;
+  using AbstractBase::m_permeability;
+  using AbstractBase::m_dPerm_dPres;
+
+  using AbstractBase::m_localMatrix;
+  using AbstractBase::m_localRhs;
+
+  using Base = singlePhaseFVMKernels::DirichletFaceBasedAssemblyKernel< NUM_DOF, FLUIDWRAPPER >;
+  using Base::numDof;
+  using Base::numEqn;
+  using Base::m_stencilWrapper;
+  using Base::m_seri;
+  using Base::m_sesri;
+  using Base::m_sei;
+  using Base::m_facePres;
+  using Base::m_faceGravCoef;
+
+  using ThermalSinglePhaseFlowAccessors =
+    StencilAccessors< fields::flow::temperature,
+                      fields::flow::dMobility_dPressure,
+                      fields::flow::dMobility_dTemperature >;
+
+  using ThermalSinglePhaseFluidAccessors =
+    StencilMaterialAccessors< SingleFluidBase,
+                              fields::singlefluid::dDensity_dTemperature,
+                              fields::singlefluid::enthalpy,
+                              fields::singlefluid::dEnthalpy_dPressure,
+                              fields::singlefluid::dEnthalpy_dTemperature >;
+
+  using ThermalConductivityAccessors =
+    StencilMaterialAccessors< SinglePhaseThermalConductivityBase,
+                              fields::thermalconductivity::effectiveConductivity >;
+
+  /**
+   * @brief Constructor for the kernel interface
+   * @param[in] rankOffset the offset of my MPI rank
+   * @param[in] faceManager the face manager
+   * @param[in] stencilWrapper reference to the stencil wrapper
+   * @param[in] fluidWrapper reference to the fluid wrapper
+   * @param[in] dofNumberAccessor
+   * @param[in] singlePhaseFlowAccessors
+   * @param[in] singlePhaseFluidAccessors
+   * @param[in] permeabilityAccessors
+   * @param[in] dt time step size
+   * @param[inout] localMatrix the local CRS matrix
+   * @param[inout] localRhs the local right-hand side vector
+   */
```

Comment:
If you want you can update the Doxygen here since it seems copy-pasted from the isothermal case

Reply:
done